spec/controllers/idv/confirmations_controller_spec.rb

```diff
@@ -25,34 +27,60 @@
     end
 
     describe 'all questions answered correctly' do
-      it 'shows success' do
+      it 'updates profile and redirects to original SAML Authn request' do
+        user.identities.create(
+          last_authenticated_at: Time.current,
+          service_provider: sp1_saml_settings.issuer
+        )
+        subject.session[:saml_request_url] = sp1_authnrequest
         complete_idv_session(true)
 
         get :index
 
-        expect(response.status).to eq 200
-        expect(response.body).to include(t('idv.titles.complete'))
+        expect(response).to redirect_to subject.session[:saml_request_url]
+        expect(flash[:success]).to eq(t('idv.titles.complete'))
 
         profile.reload
 
         expect(profile).to be_active
         expect(profile).to be_verified
+
+        # PII removed from session
+        expect(subject.user_session[:idv][:applicant]).to eq nil
+      end
+
+      it 'updates profile and redirects to IdP profile when original request is missing' do
+        user.identities.create(
+          last_authenticated_at: Time.current,
+          service_provider: sp1_saml_settings.issuer
+        )
+        subject.session[:saml_request_url] = nil
+        complete_idv_session(true)
+
+        get :index
+
+        expect(response).to redirect_to(profile_index_path)
+
+        profile.reload
+
+        expect(profile).to be_active
+        expect(profile).to be_verified
+
+        # PII removed from session
+        expect(subject.user_session[:idv][:applicant]).to eq nil
       end
     end
 
     describe 'some answers incorrect' do
-      it 'shows error' do
+      it 'shows error, does not store PII' do
         complete_idv_session(false)
 
         get :index
 
         expect(response.status).to eq 200
         expect(response.body).to include(t('idv.titles.hardfail'))
 
-        profile.reload
-
-        expect(profile).to_not be_active
-        expect(profile).to_not be_verified
+        expect { Profile.find(profile.id) }.to raise_error ActiveRecord::RecordNotFound
```

Comment:
I suppose something like:

`expect(Profile.where(id: profile.id).count).to eq 0`

would do the same thing.

I like exceptions though; they are a very efficient way to provide named error cases. We're not trying to use exceptions for flow control (http://c2.com/cgi/wiki?DontUseExceptionsForFlowControl) just to verify that Something Happened. The test says "does not store PII" and the app verifies that with RecordNotFound.

Happy to change, though if we're trying to avoid `raise_error` in specs we should probably audit for consistency.